pkg/upgrade/upgrade.go

```diff
@@ -0,0 +1,21 @@
+package upgrade
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+
+	"github.com/openshift/rosa/pkg/machinepool"
+)
+
+func NewUpgradeArgsFunction() func(cmd *cobra.Command, argv []string) error {
+	return func(cmd *cobra.Command, argv []string) error {
+		flags := cmd.Flags()
+		machinepoolID, _ := flags.GetString("machinepool")
+
+		if !machinepool.MachinePoolKeyRE.MatchString(machinepoolID) {
+			return fmt.Errorf("expected a valid identifier for the machine pool")
```

Comment:
@Mischulee We have this string repeated 3 times in this PR, so I'd suggest to make it a `const` in the `upgrade` package.

Reply:
Alternatively you could make the `error` itself a package-level var and return that e.g:

```
var (
	InvalidMachinePoolIdentifier = fmt.Errorf("expected a valid identifier for the machine pool")
)
```

```
if !machinepool.MachinePoolKeyRE.MatchString(machinepoolID) {
			return InvalidMachinePoolIdentifier
		}
```

Reply:
You are right, I added the second option.